http/dotnet/httpclient/Microsoft.Kiota.Http.HttpClient.Tests/Middleware/RedirectHandlerTests.cs

```diff
@@ -0,0 +1,221 @@
+            Assert.NotSame(response.RequestMessage, httpRequestMessage);
+            Assert.NotNull(response.RequestMessage?.Content);
+            Assert.Equal("Hello World", await response.RequestMessage.Content.ReadAsStringAsync());
+
+        }
+
+        [Fact]
+        public async Task ShouldRedirectChangeMethodAndContent()
+        {
+            // Arrange
+            var httpRequestMessage = new HttpRequestMessage(HttpMethod.Post, "http://example.org/foo")
+            {
+                Content = new StringContent("Hello World")
+            };
+            var redirectResponse = new HttpResponseMessage(HttpStatusCode.SeeOther);
+            redirectResponse.Headers.Location = new Uri("http://example.org/bar");
+            this._testHttpMessageHandler.SetHttpResponse(redirectResponse, new HttpResponseMessage(HttpStatusCode.OK));// sets the mock response
+            // Act
+            var response = await _invoker.SendAsync(httpRequestMessage, new CancellationToken());
+            // Assert
+            Assert.NotEqual(response.RequestMessage?.Method, httpRequestMessage.Method);
+            Assert.Equal(response.RequestMessage?.Method, HttpMethod.Get);
+            Assert.NotSame(response.RequestMessage, httpRequestMessage);
+            Assert.Null(response.RequestMessage?.Content);
+        }
+
+        [Theory]
+        [InlineData(HttpStatusCode.MovedPermanently)]  // 301
+        [InlineData(HttpStatusCode.Found)]  // 302
+        [InlineData(HttpStatusCode.TemporaryRedirect)]  // 307
+        [InlineData((HttpStatusCode)308)] // 308
+        public async Task RedirectWithDifferentHostShouldRemoveAuthHeader(HttpStatusCode statusCode)
+        {
+            // Arrange
+            var httpRequestMessage = new HttpRequestMessage(HttpMethod.Get, "http://example.org/foo");
+            httpRequestMessage.Headers.Authorization = new AuthenticationHeaderValue("fooAuth", "aparam");
+            var redirectResponse = new HttpResponseMessage(statusCode);
+            redirectResponse.Headers.Location = new Uri("http://example.net/bar");
+            this._testHttpMessageHandler.SetHttpResponse(redirectResponse, new HttpResponseMessage(HttpStatusCode.OK));// sets the mock response
+            // Act
+            var response = await _invoker.SendAsync(httpRequestMessage, new CancellationToken());
+            // Assert
+            Assert.NotSame(response.RequestMessage, httpRequestMessage);
+            Assert.NotSame(response.RequestMessage?.RequestUri?.Host, httpRequestMessage.RequestUri?.Host);
+            Assert.Null(response.RequestMessage?.Headers.Authorization);
+        }
+
+        [Theory]
+        [InlineData(HttpStatusCode.MovedPermanently)]  // 301
+        [InlineData(HttpStatusCode.Found)]  // 302
+        [InlineData(HttpStatusCode.TemporaryRedirect)]  // 307
+        [InlineData((HttpStatusCode)308)] // 308
+        public async Task RedirectWithDifferentSchemeShouldRemoveAuthHeader(HttpStatusCode statusCode)
```

Comment:
I really see the value of for the first point. I have made an update via https://github.com/microsoft/kiota/pull/589/commits/fee7d2d33b3c5eab95113a1cfa1f68cc70430540 to add a `AllowRedirectOnSchemeChange` property in the `RedirectHandlerOption` which defaults to false so that users will have to opt in to allow this.

For the Blazor scenario, I'm not really sure we want to go that far as of yet. From the [documentation](https://docs.microsoft.com/en-us/aspnet/core/blazor/security/webassembly/graph-api?view=aspnetcore-5.0)(for graph at least), it looks like the requests configured clients are setup by setting the Auth header.

Reply:
I'm okay shelving 2 for now.